File: go/bob/bob.go
// This is a "stub" file.  It's a little start on your solution.
// It's not a complete solution though; you have to write some code.

// Package bob should have a package comment that summarizes what it's about.
// https://golang.org/doc/effective_go.html#commentary
package bob

import "strings"

// Hey should have a comment documenting it.
func Hey(remark string) string {
	var result string
	remark = strings.TrimSpace(remark)

	switch {
	case isShouting(remark) && isQuestioning(remark):
		result = "Calm down, I know what I'm doing!"
	case isSilence(remark):
		result = "Fine. Be that way!"
	case isShouting(remark):
		result = "Whoa, chill out!"
	case isQuestioning(remark):
		result = "Sure."
	default:
		result = "Whatever."
	}

	return result
}

func isSilence(s string) bool {
	return len(s) == 0
}

func isQuestioning(s string) bool {
	return strings.HasSuffix(s, "?")
}

func isShouting(s string) bool {
	return s == strings.ToUpper(s) && s != strings.ToLower(s)
}

File: swift/bob/Sources/Bob/Bob.swift
import Foundation

class Bob {
    static func hey(_ input:String) -> String {
        switch true {
            case isShouting(input): return "Whoa, chill out!"
            case isSilence(input): return "Fine. Be that way!"
            case isAskingQuestion(input): return "Sure."
            default: return "Whatever."
        }
    }

    private static func isShouting(_ input:String) -> Bool {
        let hasLowercaseLetters = input != input.uppercased()
        let hasUppercaseLetters = input != input.lowercased()
        return hasUppercaseLetters && !hasLowercaseLetters
    }
    
    private static func isSilence(_ input:String) -> Bool {
        let inputWithoutWhitespace = input.trimmingCharacters(in: .whitespacesAndNewlines)
        return inputWithoutWhitespace == ""
    }
    
    private static func isAskingQuestion(_ input:String) -> Bool {
        return input.hasSuffix("?")
    }
}